### ISSUE TITLE
Add more configuration options for `InterfaceTypeParameterName`

### ISSUE THREAD

**Issue by jxblum** [checkstyle/checkstyle#12427]
One of my interfaces was flagged with this error, having a signature:

```java
interface MyType<ID extends Comparable<ID>>  {
    // ...
}
```

I believe it is because my type variable `ID` "`extends Comparable<ID>`" thereby not matching the default, configured format.

I like this check because I do want `ID` to be restricted by the format pattern, but effectively everything after "ID" should be ignored up to the next comma, or `>`.

A slightly more complex situation:

```java
interface AnotherType<ID extends Comparable<ID>, USER, T>
```

And so on.

I suppose I could also tweak the REGEX myself, but not currently sure how I would go about it.

**Comment by rnveach** [comment]
@jxblum Please update your first post to have more details we ask for at https://checkstyle.org/report_issue.html#How_to_request_new_feature_for_existing_functionality.3F .

**Comment by jxblum** [comment]
@rnveach - I can specify more details after awhile. In summary, I'd say that the default format pattern ("`^[A-Z]$`") for the `InterfaceTypeParameterName` check is insufficient, but arguable a developer can also customize it.

In general type variables can get quite complex in their definitions, for example...

```java
public interface Bean<ID extends Comparable<ID>, USER, PROCESS> extends Auditable<USER, PROCESS, ID>,
  Cloneable, Comparable<Bean<ID, USER, PROCESS>>, Serializable, Visitable {
```

I am pretty sure this check would fail for multiple reasons in the interface definition defined above.

Anyway, my ask was just a suggestion, and I figured the Checkstyle development team might have some other ideas on how to make the configuration of the `InterfaceTypeParameterName` check more flexible (e.g. such as ignoring certain types) since defining REGEX to cover the interface above is rather tricky.

More to follow...

**Comment by jxblum** [comment]
Actually, I think I could be complete wrong in my argument here and simply need to configure the `format` as "`^[A-Z]+$`" since most of my type variables are not a single letter.

In other words, I don't think the `InterfaceTypeParameterName` check cares that a type parameters "extends" another type that does not match the pattern.

**Comment by jxblum** [comment]
Damn! That was totally it. Sorry for the noise.
